tests/schema/system/slurm/strategy/test_slurm_command_gen_strategy.py

```diff
@@ -0,0 +1,44 @@
+from unittest.mock import mock_open, patch
+
+import pytest
+from cloudai.schema.system import SlurmSystem
+from cloudai.schema.system.slurm import SlurmNode, SlurmNodeState
+from cloudai.schema.system.slurm.strategy import SlurmCommandGenStrategy
+
+
+@pytest.fixture
+def strategy_fixture():
+    slurm_system = SlurmSystem(
+        name="TestSystem",
+        install_path="/path/to/install",
+        output_path="/path/to/output",
+        default_partition="main",
+        partitions={"main": [SlurmNode(name="node1", partition="main", state=SlurmNodeState.IDLE)]},
+    )
+    env_vars = {"TEST_VAR": "VALUE"}
+    cmd_args = {"test_arg": "test_value"}
+    strategy = SlurmCommandGenStrategy(slurm_system, env_vars, cmd_args)
+    return strategy
+
+
+@pytest.mark.usefixtures("strategy_fixture")
```

Comment:
[minor] You don't need `@pytest.mark.usefixtures("strategy_fixture")` since you are explicitly referencing `strategy_fixture` in the test function args.

I think we can make this test simpler:
1. Do not mock `open` and `os.path.join`. Since we can pass `output_path`, we can utilise `pytest`'s internal fixture `tmppath: Path`. You can convert it into a string with `str(tmppath)`.
2. After calling `_write_sbatch_script`, check that required file exists under `tmppath`.
3. Additionally, you can also check its content. At least for the arguments you are passing into the function.

Reply:
That makes sense. Thank you! Updated. I wanted to check the file name only because this pull request concerns the filename.